### PR TITLE
PowApprox mutable internal logic

### PR DESCRIPTION
**Pull request by mconcat**
closes: #56

### CODE REVIEW

File: x/gamm/keeper/math.go [hunk 1]
```diff
@@ -294,18 +294,21 @@ func powApprox(base sdk.Dec, exp sdk.Dec, precision sdk.Dec) sdk.Dec {
...
+	a := exp.Clone()
+	bigK := sdk.NewDec(0)
 	// TODO: Document this computation via taylor expansion
-	for i := 1; term.GTE(precision); i++ {
-		bigK := sdk.OneDec().MulInt64(int64(i))
-		c, cneg := absDifferenceWithSign(a, bigK.Sub(one))
-		term = term.Mul(c.Mul(x))
-		term = term.Quo(bigK)
+	for i := int64(1); term.GTE(precision); i++ {
+		a.Set(exp)
```
Comment (ValarDragon) [inline x/gamm/keeper/math.go:308]: whats going on with `a.Set(exp)` and `bigK`. These don't seem the same across the update

Comment (mconcat) [reply x/gamm/keeper/math.go:308]: I thought `absDifferenceWithSign` was mutable but it isn't, will fix this.

Comment (mconcat) [reply x/gamm/keeper/math.go]: Actually we could gain speed if we make `abs` mutable.

**Comment by mconcat** [comment]
`mconcat/56-powapprox-opt`
```
goos: darwin
goarch: amd64
pkg: github.com/osmosis-labs/osmosis/x/gamm/keeper
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkPow                  33          37631000 ns/op         4247544 B/op     191873 allocs/op
BenchmarkPow-2                32          35742374 ns/op         4247695 B/op     191875 allocs/op
BenchmarkPow-4                33          37513079 ns/op         4247849 B/op     191875 allocs/op
BenchmarkPow-8                28          36158232 ns/op         4248223 B/op     191878 allocs/op
BenchmarkPow-16               32          35642177 ns/op         4248668 B/op     191878 allocs/op
BenchmarkSqrtPow           95598             12724 ns/op            3624 B/op        199 allocs/op
BenchmarkSqrtPow-2        101456             11737 ns/op            3624 B/op        199 allocs/op
BenchmarkSqrtPow-4         90078             11598 ns/op            3624 B/op        199 allocs/op
BenchmarkSqrtPow-8        102190             11701 ns/op            3624 B/op        199 allocs/op
BenchmarkSqrtPow-16       101715             11598 ns/op            3624 B/op        199 allocs/op
```

`main`
```go
goos: darwin
goarch: amd64
pkg: github.com/osmosis-labs/osmosis/x/gamm/keeper
cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
BenchmarkPow                  20          67177389 ns/op        33736621 B/op     970000 allocs/op
BenchmarkPow-2                19          59532866 ns/op        33737600 B/op     970008 allocs/op
BenchmarkPow-4                19          57907179 ns/op        33739167 B/op     970019 allocs/op
BenchmarkPow-8                20          58669216 ns/op        33741359 B/op     970028 allocs/op
BenchmarkPow-16               19          58099349 ns/op        33745091 B/op     970036 allocs/op
BenchmarkSqrtPow           64537             16292 ns/op            7384 B/op        309 allocs/op
BenchmarkSqrtPow-2         80757             14496 ns/op            7384 B/op        309 allocs/op
BenchmarkSqrtPow-4         78423             15511 ns/op            7384 B/op        309 allocs/op
BenchmarkSqrtPow-8         78834             14515 ns/op            7384 B/op        309 allocs/op
BenchmarkSqrtPow-16        83053             14379 ns/op            7384 B/op        309 allocs/op
```

**Comment by ValarDragon** [comment]
Awesome, so it is like a 2x speed improvement!(Also the thing with heap improvements is that they are more impactful as he overall memory situation gets worse.

Lets sync with Sunny whether this gets merged today, or like this weekend in a version that just makes nodes faster

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**